readthedocs/api/v2/tasks.py

```diff
@@ -0,0 +1,29 @@
+from datetime import timedelta
+
+import structlog
+from django.utils import timezone
+
+from readthedocs.api.v2.models import BuildAPIKey
+from readthedocs.worker import app
+
+log = structlog.get_logger(__name__)
+
+
+@app.task(queue="web")
+def delete_old_revoked_build_api_keys():
+    """
+    Delete revoked and expired keys that are older than 7 days.
+
+    We don't delete keys created in the last 7 days,
+    to have some audit trail in case we need to investigate something.
+    """
+    created_before = timezone.now() - timedelta(days=7)
+    to_delete = BuildAPIKey.objects.filter(revoked=True, created__lt=created_before)
```

Comment:
I know in the past these queries have been too slow and gotten cancelled, and we've moved to raw SQL. I guess we can try this and migrate if it's too slow?